mitodl/open-discussions
# CELERY_WORKER_MAX_MEMORY_PER_CHILD setting

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes https://github.com/mitodl/open-discussions/issues/3251

#### What's this PR do?
Adds `CELERY_WORKER_MAX_MEMORY_PER_CHILD` as a django setting, configurable via .env, with a default of 250 MB.

#### How should this be manually tested?
- On master, run `recreate_index`.  In another terminal run `docker stats open-discussions_celery_1` and monitor the memory usage.
- Switch to this branch, optionally set add `CELERY_WORKER_MAX_MEMORY_PER_CHILD=50000` to your .env file, then run `recreate_index` again and monitor the memory.  The memory usage should be lower, though this may be conditional on how many objects you have that need indexing.

Once this is on heroku CI, I'll try running `recreate_index` with the default and adjust the value if necessary, to see if I can get it to complete without getting killed.
